src/core/server/base.js

```diff
@@ -262,127 +265,105 @@ function baseServer(
-          return next(matchError);
-        }
-
-        if (!renderProps) {
-          log.info(`match() did not return renderProps for ${req.url}`);
-          return showErrorPage({ createStore, status: 404, req, res, config });
-        }
+    const history = createHistory({ req });
+    const { sagaMiddleware, store } = createStore({ history });
 
-        let pageProps;
-        let runningSagas;
+    let pageProps;
+    let runningSagas;
 
-        try {
-          cookie.plugToRequest(req, res);
+    try {
+      cookie.plugToRequest(req, res);
 
-          let sagas = appSagas;
-          if (!sagas) {
-            // eslint-disable-next-line global-require, import/no-dynamic-require
-            sagas = require(`${appName}/sagas`).default;
-          }
-          runningSagas = sagaMiddleware.run(sagas);
+      let sagas = appSagas;
+      if (!sagas) {
+        // eslint-disable-next-line global-require, import/no-dynamic-require
+        sagas = require(`${appName}/sagas`).default;
+      }
+      runningSagas = sagaMiddleware.run(sagas);
 
-          // TODO: synchronize cookies with Redux store more automatically.
-          // See https://github.com/mozilla/addons-frontend/issues/5617
-          const token = cookie.load(config.get('cookieName'));
-          if (token) {
-            store.dispatch(setAuthToken(token));
-          }
-          if (
-            cookie.load(config.get('dismissedExperienceSurveyCookieName')) !==
-            undefined
-          ) {
-            store.dispatch(dismissSurvey());
-          }
+      // TODO: synchronize cookies with Redux store more automatically.
+      // See https://github.com/mozilla/addons-frontend/issues/5617
+      const token = cookie.load(config.get('cookieName'));
+      if (token) {
+        store.dispatch(setAuthToken(token));
+      }
+      if (
+        cookie.load(config.get('dismissedExperienceSurveyCookieName')) !==
+        undefined
+      ) {
+        store.dispatch(dismissSurvey());
+      }
 
-          pageProps = getPageProps({ noScriptStyles, store, req, res, config });
+      pageProps = getPageProps({ noScriptStyles, store, req, res, config });
 
-          if (config.get('disableSSR') === true) {
-            // This stops all running sagas.
-            store.dispatch(END);
+      if (config.get('disableSSR') === true) {
+        // This stops all running sagas.
+        store.dispatch(END);
 
-            await runningSagas.done;
-            log.warn('Server side rendering is disabled.');
+        await runningSagas.done;
+        log.warn('Server side rendering is disabled.');
 
-            return hydrateOnClient({ res, pageProps });
-          }
-        } catch (preLoadError) {
-          log.info(oneLine`Caught an error in match() before rendering:
+        return hydrateOnClient({ res, pageProps });
+      }
+    } catch (preLoadError) {
+      log.info(oneLine`Caught an error in match() before rendering:
```

Comment:
It might help avoid confusion if you update this message. It no longer runs in `match()`.

Reply:
ah oops.